### ISSUE TITLE
Fix OpenSSL 1.1.0 incompatibilities --- bignum_st declare missing

### ISSUE THREAD

**Issue by kjx98** [exploitagency/vanitygen-plus#115]
```
diff --git a/oclengine.c b/oclengine.c
index 6585fc6..b9a0fb1 100644
--- a/oclengine.c
+++ b/oclengine.c
@@ -57,6 +57,16 @@
 #define PPR_ARROW_Y ppr->Y
 #define PPC_ARROW_X ppc->X
 #define PPC_ARROW_Y ppc->Y
+
+struct bignum_st {
+    BN_ULONG *d;                /* Pointer to an array of 'BN_BITS2' bit
+                                 * chunks. */
+    int top;                    /* Index of last used d +1. */
+    /* The next are internal book keeping for bn_expand. */
+    int dmax;                   /* Size of the d array. */
+    int neg;                    /* one if the number is negative */
+    int flags;
+};
 #else
 #define PPNT_ARROW_X &ppnt->X
 #define PPNT_ARROW_Y &ppnt->Y
diff --git a/vanitygen.c b/vanitygen.c
index 992e5d7..76ecf50 100644
--- a/vanitygen.c
+++ b/vanitygen.c
@@ -39,7 +39,18 @@ int GRSFlag = 0;
 
 const char *version = VANITYGEN_VERSION;
 
-
+// Unfortunately we need this!
+#if OPENSSL_VERSION_NUMBER >= 0x0010100000
+struct bignum_st {
+    BN_ULONG *d;                /* Pointer to an array of 'BN_BITS2' bit
+                                 * chunks. */
+    int top;                    /* Index of last used d +1. */
+    /* The next are internal book keeping for bn_expand. */
+    int dmax;                   /* Size of the d array. */
+    int neg;                    /* one if the number is negative */
+    int flags;
+};
+#endif // OPENSSL_VERSION_NUMBER
 /*
  * Address search thread main loop
  */

```

**Comment by exploitagency** [comment]
?

Please post using code tags

**Comment by exploitagency** [comment]
Duplicate of ISSUE #114 ?
Fixed in PR #116 ?

Was the below your issue?

> 
> libssl-dev:amd64 (1.1.0h-4)
> 
>     make all
>     cc -ggdb -O3 -Wall -c -o vanitygen.o vanitygen.c
>     vanitygen.c: In function ‘vg_thread_loop’:
>     vanitygen.c:132:24: error: dereferencing pointer to incomplete type ‘BIGNUM {aka struct bignum_st}’
>     memcpy((char *)pkbn->d + 32 - vcp->vc_privkey_prefix_length, vcp->vc_privkey_prefix, vcp->vc_privkey_prefix_length);
>     ^~
>     : recipe for target 'vanitygen.o' failed
>     make: *** [vanitygen.o] Error 1
>

**Comment by exploitagency** [comment]
Closing. Can request to reopen if needed
